big_tests/tests/mod_filter_SUITE.erl

```diff
@@ -0,0 +1,156 @@
+                                                                                        [{allow, all}]]),
+
+    %restriction for localhost
+    {atomic,ok} = rpc(mim(), ejabberd_config, add_global_option, [{access, restrict_dom1, global}, 
+                                                                  [{allow, domain1}, {deny, all}]]),
+    %restriction for sogndal
+    {atomic,ok} = rpc(mim(), ejabberd_config, add_global_option, [{access, restrict_dom2, global},
+                                                                  [{allow, domain2}, {deny, all}]]),
+    %restriction for bot
+    {atomic,ok} = rpc(mim(), ejabberd_config, add_global_option, [{access, restrict_bot, global}, 
+                                                                  [{allow, bot}, {deny, all}]]).
+%%--------------------------------------------------------------------
+%% Message tests
+%%--------------------------------------------------------------------
+
+allowed_messages_story(Config) ->
+    escalus:story(Config, [{alice, 1}, {bob, 1}, {kate, 1}, {jon, 1}], fun(Alice, Bob, Kate, Jon) ->
+        %according to update_rules() example, anonymous can send message to bot.
+        erlang:put(anon_user, escalus_utils:get_jid(Jon)),
+        escalus_client:send(Jon, escalus_stanza:chat_to(Kate, <<"Hi!">>)),
+        escalus:assert(is_chat_message, [<<"Hi!">>],
+                       escalus:wait_for_stanza(Kate)),
+
+        %according to update_rules() example, bot can send message to anonymous.
+        erlang:put(anon_user, escalus_utils:get_jid(Jon)),
+        escalus_client:send(Kate, escalus_stanza:chat_to(Jon, <<"Hi!">>)),
+        escalus:assert(is_chat_message, [<<"Hi!">>],
+                       escalus:wait_for_stanza(Jon)),
+
+        %according to update_rules() example, user can send message to
+        %other user within one host.
+        escalus_client:send(Alice, escalus_stanza:chat_to(Bob, <<"Hi!">>)),
+        escalus:assert(is_chat_message, [<<"Hi!">>],
+                       escalus:wait_for_stanza(Bob)),
+
+        %according to update_rules() example, user can send message to bot.
+        escalus_client:send(Alice, escalus_stanza:chat_to(Kate, <<"Hi!">>)),
+        escalus:assert(is_chat_message, [<<"Hi!">>],
+                       escalus:wait_for_stanza(Kate)),
+    
+        %according to update_rules() example, bot can send message to user.
+        escalus_client:send(Kate, escalus_stanza:chat_to(Alice, <<"Hi!">>)),
+        escalus:assert(is_chat_message, [<<"Hi!">>],
+                       escalus:wait_for_stanza(Alice))
+    end).
+
+denied_messages_story(Config) ->
+    escalus:story(Config, [{alice, 1}, {astrid, 1}, {bob, 1}, {jon, 1}], fun(Alice, Astrid, Bob, Jon) ->
+        erlang:put(anon_user, escalus_utils:get_jid(Jon)),
+
+        %according to update_rules() example, anonymous can't send message to user.
+        escalus_client:send(Jon, escalus_stanza:chat_to(Bob, <<"Hi!">>)),
```

Comment:
The same here as https://github.com/esl/MongooseIM/pull/2032/files#diff-b09a62b822afb1566dc112a2aa8fb3b1R102. We can extract a common part.